### PR TITLE
Add EIP: Separate gas type for calldata

### PR DESCRIPTION
**Pull request by vbuterin**
Add a new type of gas for transaction calldata. Add a new transaction type that provides max_basefee and priority_fee as a vector, providing values for execution gas, blob gas and calldata gas. Modify EIP-1559 to use the same mechanism for the three types of gas.

### CODE REVIEW

File: EIPS/eip-7704.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+---
+eip: 7704
```
Comment (abcoathup) [inline EIPS/eip-7704.md:2]:
```suggestion
eip: 7706
```
Assigning next sequential EIP/ERC/RIP number.  7704 had already been assigned to an ERC.

Please also update the filename

File: EIPS/eip-7704.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+title: Separate gas type for calldata
+description: Create a separate basefee and gaslimit for calldata
+author: Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/create-a-separate-basefee-and-gaslimit-for-calldata/19998
```
Comment (abcoathup) [inline EIPS/eip-7704.md:6]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7706-create-a-separate-basefee-and-gaslimit-for-calldata/19998
```

File: EIPS/eip-7706.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+We add the functions `get_max_fees` and `get_priority_fees`, to compute these length-3 vectors for previous transaction types:
+
+```
+def get_max_fees(tx: Transaction) -> [int, int, int]:
+    if tx.type == NEW_TX_TYPE:
+        return tx.max_fees_per_gas
+    elif tx.type == BLOB_TX_TYPE:
+        return [tx.max_fee_per_gas, tx.max_fee_per_blob_gas, tx.max_fee_per_gas]
+    elif is_eip_1559(tx.type):
+        return [tx.max_fee_per_gas, 0, max_fee_per_gas]
```
Comment (etan-status) [inline EIPS/eip-7706.md:66]:
```suggestion
        return [tx.max_fee_per_gas, 0, tx.max_fee_per_gas]
```

**Review by g11tech — APPROVED** [review]

**Review by VersoriumX — COMMENTED** [review]
VersoriumX

File: EIPS/eip-7706.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ def get_max_fees(tx: Transaction) -> [int, int, int]:
     elif tx.type == BLOB_TX_TYPE:
         return [tx.max_fee_per_gas, tx.max_fee_per_blob_gas, tx.max_fee_per_gas]
     elif is_eip_1559(tx.type):
-        return [tx.max_fee_per_gas, 0, max_fee_per_gas]
```
Comment (VersoriumX) [inline EIPS/eip-7706.md]: `commit 9eb41928548665e15617b27d258ac13379390acc`

**Pull request merged**